Libraries/CO.CDP.WebApi.Foundation/ResponseMiddleware.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace CO.CDP.WebApi.Foundation;
+
+internal class ResponseMiddleware(
+        RequestDelegate next,
+        ILogger<ResponseMiddleware> logger,
+        Dictionary<Type, (int, string)> exceptionMap)
+{
+    public async Task Invoke(HttpContext context)
+    {
+        var requestUrl = context.Request.GetDisplayUrl();
+
+        try
+        {
+            await next.Invoke(context);
+
+            var statusCode = context.Response.StatusCode;
+
+            if (statusCode >= 400 && statusCode < 500)
+            {
+                logger.LogInformation("Response status: {statusCode}, for request: {requestUrl}", statusCode, requestUrl);
+
+                await HandleResponse(context, statusCode, new ProblemDetails { Status = statusCode });
+            }
+        }
+        catch (Exception ex)
+        {
+            var statusCode = StatusCodes.Status500InternalServerError;
+            var errorCode = "GENERIC_ERROR";
+            var pd = new ProblemDetails { Status = statusCode };
```

Comment:
Since we create a new instance of ProblemDetails here (we didn't prteviously), do we lose the default behaviour? Or maybe, problem details created here are merged with the defaults?

Reply:
Yes that's the problem, and I am looking over internet for a solution.
And hence the draft PR :)

Reply:
Oh I see it contains default behaviour, like setting title